### ISSUE TITLE
About online clustering(sinkhorn iter algorithm)

### ISSUE THREAD

**Issue by seonhoon1002** [FengZicai/Cluster3Dseg#5]
Thanks for releasing the code.

I have some questions about sinkhorn alogrithm in your code. 

        r = inv_K / (self.PS @ c)          # (KxN)@(N,1) = K x 1
        c_new = inv_N / (r.T @ self.PS).T  # ((1,K)@(KxN)).t() = N x 1
        if _counter % 10 == 0:
            err = torch.abs((c.squeeze() / c_new.squeeze()) - ones)
            err = torch.sum(torch.where(torch.isnan(err), torch.full_like(err, 0), err))

The line 23,24 in "spvnas>cluster_contrastloss>online_clustering", It seems like inv_K and inv_N are scala, which are originally marginal vectors in Sinhorn Distance(Neurips@2013).

I think marginal vectors have important roles in sinkhorn algorithms, but in your code, they are just inverse size of subclass and points.

So can I ask why inv_K and inv_N are scala value not marginal vectors?

**Comment by FengZicai** [comment]
r and c are the so-called marginal projections of the matrix onto its rows and columns, respectively (page 3 of SELF-LABELLING VIA SIMULTANEOUS CLUSTERING AND REPRESENTATION LEARNING).

You can refer to https://openreview.net/pdf?id=Hyx-jyBFPr and https://github.com/yukimasano/self-label for details.

**Comment by seonhoon1002** [comment]
I think I'll need more time to verify them but It seems like that you're right.

Thank you for your fast comment.
